### PR TITLE
Plfm 2730

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/table/AsynchDownloadRequestBody.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+			"type": "string",
+			"description": "The SQL to use to build the resulting CSV."
+		},
+		"writeHeader": {
+			"type": "boolean",
+			"description": "Should the first line contain the columns names as a header in the resulting file?  Set to 'true' to include the headers else, 'false'.  The default value is 'true'."
+		},
+		"separator": {
+			"type": "string",
+			"description": "The delimiter to used for separating entries in the resulting file. The default character ',' will be used if this is not provided by the caller.  For tab-separated values use '\\t'"
```
Comment (marcel-blonk) [inline lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/table/AsynchDownloadRequestBody.json:19]: to be used

File: lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/table/AsynchDownloadResponseBody.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		"tableId": {
+		    "description":"The ID of the table identified in the from clause of the table query.",
+			"type": "string"
+		},
+		"etag": {
+		    "description":"Any RowSet returned from Synapse will contain the current etag of the change set.  To update any rows from a RowSet the etag must be provided with the POST.",
+			"type": "string"
+		},
+		"headers": {
+		    "description":"The list of ColumnModels ID that describes the rows of this set.",
```
Comment (marcel-blonk) [inline lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/table/AsynchDownloadResponseBody.json:22]: ColumnModel IDs

File: lib/lib-worker/src/main/java/org/sagebionetworks/asynchronous/workers/sqs/MessageReceiverImpl.java [hunk 3]
```diff
@@ -275,32 +257,63 @@ private int triggerFiredImpl() throws InterruptedException {
...
+	 * @param toBeProcessed
+	 * @param workerProgresss
+	 * @return
+	 */
+	public List<WorkerData> startWorkers(List<Message> toBeProcessed,
+			WorkerProgress workerProgresss) {
+		List<WorkerData> currentWorkers = new LinkedList<WorkerData>();
+		List<Message> messageBatch = new LinkedList<Message>();
+		int workerId = 0;
+		for (Message message: toBeProcessed) {
```
Comment (marcel-blonk) [inline lib/lib-worker/src/main/java/org/sagebionetworks/asynchronous/workers/sqs/MessageReceiverImpl.java:286]: Lists.partition?

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/MultipartTransferListener.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package org.sagebionetworks.repo.manager.file;
```
Comment (marcel-blonk) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/MultipartTransferListener.java:1]: dead code

File: services/workers/src/main/java/org/sagebionetworks/table/worker/TableCSVDownloadWorker.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+			// This just means we cannot do this right now.  We can try again later.
+			asynchJobStatusManager.updateJobProgress(status.getJobId(), 0L, 100L, "Waiting for the table index to become available...");
+			// do not return the message because we do not want it to be deleted.
+			return null;
+		}catch(Throwable e){
+			// The job failed
+			asynchJobStatusManager.setJobFailed(status.getJobId(), e);
+			throw e;
+		}finally{
+			if(writer != null){
```
Comment (marcel-blonk) [inline services/workers/src/main/java/org/sagebionetworks/table/worker/TableCSVDownloadWorker.java:91]: IOUtils.closeQuietly

Comment (john-hill) [inline services/workers/src/main/java/org/sagebionetworks/table/worker/TableCSVDownloadWorker.java:91]: Could not use in this case as CSVWriter does not extend writer.

**Pull request merged**